src/index.js

```diff
@@ -1,10 +1,14 @@
 // @flow
 import React from "react";
 
+declare function __webpack_require__(pathOrId: string | number): any;
 type GenericComponent<Props> = Class<React.Component<{}, Props, mixed>>;
 type LoadedComponent<Props> = GenericComponent<Props>;
 type LoadingComponent = GenericComponent<{}>;
 
+let isWebpack = typeof __webpack_require__ !== "undefined";
+let requireFn = isWebpack ? __webpack_require__ : module.require;
```

Comment:
I'm having trouble seeing how this could work, since `Module.prototype.require` refers to `this` expecting it to be the `module` object, and is never bound to that object (not even [here](https://github.com/nodejs/node/blob/5bfd13b81e38b60a7b9f346fbfcb216192cf0974/lib/module.js#L582), in case you had a similar hope).

Unless you have an explanation of how this works, I have to insist that it's dangerous to call `module.require` without knowing what `module.id` (the parent module identifier) is.

If it's not clear why this works (if it does), please just bind it to be safe!